ambry-network/src/main/java/com.github.ambry.network/SocketServer.java

```diff
@@ -325,13 +320,8 @@ public void run() {
   /*
    * Create a server socket to listen for connections on.
    */
-  private ServerSocketChannel openServerSocket(String host, int port) throws IOException {
-    InetSocketAddress address = null;
-    if (host == null || host.trim().isEmpty()) {
-      address = new InetSocketAddress(port);
-    } else {
-      address = new InetSocketAddress(host, port);
```

Comment:
Yes, this is the place. B/C we usually specify a fully qualified domain name, it does a lookup and only binds for an IPV4 address. I guess java's networking libraries work on both if you do not specify your hostname (which we don't need to do, since this will always be a local address)